back/boxtribute_server/models/crud.py

```diff
@@ -65,6 +80,170 @@ def create_box(
     raise BoxCreationFailed()
 
 
+def move_items_from_box_to_distribution_event(
+    user_id, box_label_identifier, distribution_event_id, number_of_items
+):
+    """
+    Move items from a box to a distribution event.
+    """
+
+    with db.database.atomic():
+        # Completed Events should not be mutable anymore
+        distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+        if distribution_event.state == DistributionEventState.Completed:
+            raise ModifyCompletedDistributionEvent(
+                desired_operation="add_items",
+                distribution_event_id=distribution_event.id,
+            )
+
+        box = Box.get(Box.label_identifier == box_label_identifier)
+
+        if box.items < number_of_items:
+            raise NotEnoughItemsInBox(
+                box_label_identifier=box_label_identifier,
+                number_of_requested_items=number_of_items,
+                number_of_actual_items=box.items,
+            )
+
+        unboxed_items_collection, _ = UnboxedItemsCollection.get_or_create(
+            distribution_event=distribution_event_id,
+            product=box.product.id,
+            size=box.size.id,
+            defaults={"number_of_items": 0, "location": box.location.id},
+        )
+
+        unboxed_items_collection.number_of_items += number_of_items
+        box.items -= number_of_items
+
+        unboxed_items_collection.save()
+        box.save()
+        return unboxed_items_collection
+
+
+def move_box_to_distribution_event(box_label_identifier, distribution_event_id):
+    """Move a box to a distribution event."""
+    with db.database.atomic():
+        box = Box.get(Box.label_identifier == box_label_identifier)
+        distribution_event = DistributionEvent.get_by_id(distribution_event_id)
```

Comment:
Same line as 129 :) also you could move the box query to line 134 because in the case of an error it's superfluous

Reply:
removed the duplicated line :D 
but regarding moving it to after the error checking: the distribution_event is needed for identifying whether there is an error (it's checking the state of the event).

Reply:
I meant the line `box = Box.get(...)` can be moved because it's not require for the state validation